src/state.rs

```diff
@@ -806,6 +808,61 @@ mod tests {
         assert_eq!(secs_to_duration(secs), duration);
     }
 
+    #[test]
+    fn test_duration_after_finish_and_leave() {
+        let duration = Duration::from_secs(42);
+        let now = Instant::now();
+
+        let mut state = state_started_at(now - duration);
+        state.finish_using_style(now, ProgressFinish::AndLeave);
+
+        assert_eq!(
+            state.state.duration(),
+            duration,
+            "Expected duration: {}, actual: {}",
+            duration_to_secs(duration),
+            duration_to_secs(state.state.duration())
+        );
+    }
+
+    #[test]
+    fn test_duration_after_finish_and_clear() {
+        let duration = Duration::from_secs(42);
+        let now = Instant::now();
+
+        let mut state = state_started_at(now - duration);
+        state.finish_using_style(now, ProgressFinish::AndClear);
+
+        assert_eq!(
+            state.state.duration(),
+            duration,
+            "Expected duration: {}, actual: {}",
+            duration_to_secs(duration),
+            duration_to_secs(state.state.duration())
+        );
+    }
+
+    fn state_started_at(start_at: Instant) -> BarState {
```

Comment:
Nit: rename the argument to `started` so that it doesn't need to be renamed in the `ProgressState` initializer.